hera_cal/abscal.py

```diff
@@ -78,7 +78,7 @@ class AbsCal(object):
     """
     def __init__(self, model, data, refant=None, wgts=None, antpos=None, freqs=None,
```

Comment:
As a general comment, I find it odd that we support so many different ways of instantiating this object and it's very difficult to keep track of how weights and flags are propagated. Also, it's pretty strange that a user would specify the data as a DataContainer but the calibration as a file path or vice versa.

Perhaps this can wait for some future analysis overhaul/streamlining, but I think we can make the code a lot more readable and maintainable if we impose consistent interfaces externally. For example, either it's an abscal object's responsibility to figure out its own weights or it's not.

Reply:
Yeah